buildenv/jenkins/jobs/builds/Build-Test-Any-Platform

```diff
@@ -0,0 +1,73 @@
+/*******************************************************************************
+ * Copyright (c) 2018, 2018 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+ARGS = ['SDK_VERSION', 'PLATFORM']
+
+node('master') {
+    timestamps {
+        try{
+            retry(5){
+                checkout scm
+            }
+
+            variableFile = load 'buildenv/jenkins/common/variables-functions'
+
+            if (!params.UPSTREAM_JOB_NAME && !params.UPSTREAM_JOB_NUMBER) {
+                variableFile.set_job_variables('build')
+                buildFile = load 'buildenv/jenkins/common/build'
+            }
+
+            if (run_tests()) {
+                variableFile.set_job_variables('test')
```

Comment:
For PR builds with testing, we will hit `set_job_variables('build')` then `set_job_variables('test')` but we should be hitting `set_job_variables('pullRequest')`. This will technically work as-is because our build/test labels (machines) are currently the same at OpenJ9. But we would essentially be setting up for a build node and then changing to a test node, which is incorrect because we need to compile on a build node. We may need another `if` here that looks for a param that is passed in a PR build.
https://ci.eclipse.org/openj9/view/Pull%20Requests/job/PullRequest-Sanity-JDK8-win_x86-64_cmprssptrs-OpenJ9/lastSuccessfulBuild/parameters/
`ghprbPullId` is usually a good one.

Reply:
added check for PR